### ISSUE TITLE
increase CI test coverage for connection target types

### ISSUE THREAD

**Issue by swanderz** [dbt-msft/dbt-sqlserver#65]
give the weird behavior we saw with connection strings and `SqlPassword=True`. @NandanHegde15 had the idea to have a separate Circle CI job would be verify that `dbt debug` works on all target configurations that are commonly used. We can either add this to #62 or in a new PR after #62 is merged.

The ones I can think of:
- [ ] `trusted_connection=True`
- [ ] `windows_integrated=True`
- [ ] when `Encrypt` and `TrustServerCertificate` aren't specified
- [ ] the various combos of`Encrypt`, `TrustServerCertificate`, and `SqlPassword`
- [ ] Active Directory
    - [ ] Password
    - [ ] Integrated (trickier as it would require a VM and an AD user in the same RG as the Azure SQL server)
    - [ ] Interactive

**Comment by mikaelene** [comment]
That would be really great. It would really help improve stability on releases to be confident that this doesn't break. I think after #62 is merged. To make the merges smaller.

**Comment by swanderz** [comment]
@jtcohen6 could you help us think how to best set up running `dbt debug` against all permutations of target configs? for example, we want to make sure that the `connections.py` logic works whether:
- the values are `true` or `false`, and if
- the params aren't specified.
```yaml
encrypt: true
trust_cert: true
```

we'd also need to do the same with all the AAD auth methods...

**Comment by jtcohen6** [comment]
@swanderz I don't think `dbt debug` returns nonzero exit codes, but you could do it just as well by executing a `dbt compile` command with each permutation.

I think you could do that outside of the pytest framework to start: create a `sample.profiles.yml` with a whole range of targets. In your Circle steps, copy that profile and execute `dbt compile --target {target}` for each.

**Comment by mikaelene** [comment]
Is anyone working on this? I would really like to push a new version to pypi. But then we need to make sure that all functionality we are shipping are working as expected...

**Comment by swanderz** [comment]
working on this today!
